### PR TITLE
Query: Relational: Initial support for DefaultIfEmpty operator.

### PR DESCRIPTION
**Pull request by anpete**
@maumar

### CODE REVIEW

File: src/EntityFramework.Relational/Query/Internal/RelationalResultOperatorHandler.cs [hunk 1]
```diff
@@ -255,65 +255,66 @@ private static Expression HandleCount(HandlerContext handlerContext)
...
 
-                return (Expression)_transformClientExpressionMethodInfo
-                    .MakeGenericMethod(minExpression.Type)
-                    .Invoke(null, new object[] { handlerContext });
-            }
+            selectExpression.ClearTables();
 
-            return handlerContext.EvalOnClient();
-        }
+            var emptySelectExpression = new SelectExpression(selectExpression.QuerySqlGeneratorFactory, "empty");
```
Comment (mikary) [inline src/EntityFramework.Relational/Query/Internal/RelationalResultOperatorHandler.cs]: Why not use `handlerContext.SelectExpressionFactory`?

File: src/EntityFramework.Relational/Query/Expressions/SelectExpression.cs [hunk 2]
```diff
@@ -57,6 +57,8 @@ public SelectExpression([NotNull] IQuerySqlGeneratorFactory querySqlGeneratorFac
             ? _projection[0].Type
             : base.Type;
 
+        public IQuerySqlGeneratorFactory QuerySqlGeneratorFactory => _querySqlGeneratorFactory;
```
Comment (mikary) [inline src/EntityFramework.Relational/Query/Expressions/SelectExpression.cs]: No need to expose the service if we use the factory in `RelationalResultOperatorHandler`

File: src/EntityFramework.Relational/Query/Internal/RelationalResultOperatorHandler.cs (hunk 1 above)
Comment (anpete) [inline src/EntityFramework.Relational/Query/Internal/RelationalResultOperatorHandler.cs]: That's better, thanks.

File: src/EntityFramework.Relational/Query/Expressions/SelectExpression.cs (hunk 2 above)
Comment (anpete) [inline src/EntityFramework.Relational/Query/Expressions/SelectExpression.cs]: Yay.

**Comment by mikary** [comment]
:shipit:

**Pull request merged**